### PR TITLE
Backport EitherT orElseFirst and orLeft

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Refs https://github.com/gcanti/fp-ts/pull/1445#issuecomment-808323185

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Haven't committed doc changes as saw failures on `E.fromPredicate` examples, which is unrelated...

**Comment by thewilkybarkid** [comment]
Just noticed the `2.11` doesn't have builds, https://github.com/gcanti/fp-ts/commit/6898b03722cddd8ed9c12469059d0f3007b8d7a4 introduced broken examples. /cc @gcanti

File: src/IOEither.ts [hunk 1]
```diff
@@ -241,6 +241,30 @@ export const orElseW: <E1, E2, B>(
...
+  /*#__PURE__*/
+  ET.orElseFirst(I.Monad)
+
+/**
+ * @category combinators
+ * @since 2.11.0
+ */
+export const orElseFirstW: <E1, E2, B>(
+  onLeft: (e: E1) => IOEither<E2, B>
+) => <A>(ma: IOEither<E1, A>) => IOEither<E1 | E2, A> = orElseFirst as any
```
Comment (thewilkybarkid) [inline src/IOEither.ts:258]: Doesn't appear to have any coverage in `3.0.0`. Will add in dtslint checks for all of these.

**Comment by gcanti** [comment]
@thewilkybarkid thanks, fixed

File: src/IOEither.ts (hunk 1 above)
Comment (gcanti) [reply src/IOEither.ts:258]: yeah, I should really scan the whole `src` and add type-level tests to each `W` function

**Comment by thewilkybarkid** [comment]
Covered the 4 categories that have it in `3.0.0` (`Either` itself and `StateReaderTaskEither` don't).

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
Thanks @thewilkybarkid :+1:

**Pull request merged**